addonscript/test/test_task.py

```diff
@@ -104,3 +117,57 @@ def test_build_filelist_missing_file(context, task_dfn):
     context.task = task_dfn
     with pytest.raises(TaskVerificationError):
         task.build_filelist(context)
+
+
+@pytest.mark.parametrize('api_root, scope, raises', (
+    ('http://localhost:5000', 'project:releng:addons.mozilla.org:server:dev', False),
+    ('http://some-amo.url', 'project:releng:addons.mozilla.org:server:dev', False),
+    ('https://addons.allizom.org', 'project:releng:addons.mozilla.org:server:staging', False),
+    ('https://addons.allizom.org/', 'project:releng:addons.mozilla.org:server:staging', False),
+    ('https://addons.mozilla.org', 'project:releng:addons.mozilla.org:server:production', False),
+    ('https://addons.mozilla.org/', 'project:releng:addons.mozilla.org:server:production', False),
+))
+def test_get_amo_instance_config_from_scope(context, api_root, scope, raises):
+    context.config['amo_instances'][scope] = copy.deepcopy(context.config['amo_instances']['project:releng:addons.mozilla.org:server:dev'])
```

Comment:
context scope is function, do we really need deepcopy?

Reply:
Good catch. Thanks!